### DISCUSSION TITLE
Hash password on client side

### DISCUSSION THREAD

**Discussion by IzacPS** [ory/kratos#973] [Q&A] [answered]
So kratos hashes de password for me in the backend. When using api with ative ui, for security, I have to hash the password in the client again, then send via api login body?

**Comment by k9ert** [comment, marked as answer]
No, i'm not sure why you might think that this is the case. Check the api-flow described here:
https://www.ory.sh/kratos/docs/self-service/#api-flows
Feel free to elaborate if that doesn't fit your use case!

**Reply by IzacPS** [reply]
Thanks for answering. What I want to now is if when developing an app natively and I use retrofit to handle de http requests, if I post the password on the body for login, will it be vulnerable?

**Reply by k9ert** [reply]
In any case, you should use ssl ~if you're worried about network security~ always. I'm not sure what kind of vulnerability you have in your mind.

**Reply by zepatrik** [reply]
It looks a bit like there are some misunderstandings about passwords and hashing @IzacPS .

> To securely manage credentials, Kratos has to only store a cryptographic hash of the credential. Whenever the credential has to be checked, the hash of the provided credential is computed and compared to the stored hash. This ensures that no one is able to retrieve the credentials, even with full access to the system's storage.

from https://www.ory.sh/choose-recommended-argon2-parameters-password-hashing/
